azure-devops/azext_devops/dev/boards/relations.py

```diff
@@ -0,0 +1,107 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from knack.util import CLIError
+from azext_devops.devops_sdk.v5_0.work_item_tracking.models import JsonPatchOperation, Wiql
+
+from azext_devops.dev.common.services import (get_work_item_tracking_client,
+                                              resolve_instance)
+
+logger = get_logger(__name__)
+
+
+def get_relation_types_show(organization=None, detect=None):
+    """ List work item relations supported in the organization.
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    client = get_work_item_tracking_client(organization)
+    return client.get_relation_types()
+
+
+def add_relation(id, relation_type, target_ids, organization=None, detect=None):  # pylint: disable=redefined-builtin
+    """ Add relation(s) to work item.
+    """
+    organization = resolve_instance(detect=detect, organization=organization)
+    patch_document = []
+    client = get_work_item_tracking_client(organization)
+
+    relation_type_system_name = get_system_relation_name(client, relation_type)
+
+    target_work_item_ids = target_ids.split(',')
+    work_item_query_clause = []
+    for target_work_item_id in target_work_item_ids:
+        work_item_query_clause.append('[System.Id] = {}'.format(target_work_item_id))
+
+    wiql_query_format = 'SELECT [System.Id] FROM WorkItems WHERE ({})'
+    wiql_query_to_get_target_work_items = wiql_query_format.format(' OR '.join(work_item_query_clause))
+
+    wiql_object = Wiql()
+    wiql_object.query = wiql_query_to_get_target_work_items
+    target_work_items = client.query_by_wiql(wiql=wiql_object).work_items
```

Comment:
For an id which does not exist this might become empty list. 
Here is the error I am getting which is not self explanatory. 

az boards work-item relation add --id 103 --relation-type 'Parent' --target-ids 106 -o json
**At least one operation is required for Apply.**

Reply:
added a simple check, not adding a very clear one because it is an error case and not the primary use case

Reply:
So what is the check and what does it show as error now?

Reply:
if any id is incorrect, we will show this
`'Id(s) supplied in --target-ids is not valid'`